pkg/token/token_test.go

```diff
@@ -38,156 +41,126 @@ func generateTestKey() (*rsa.PrivateKey, error) {
 }
 
 func TestToken(t *testing.T) {
-	issuer := "issuer"
-
-	key, err := generateTestKey()
-	if err != nil {
-		t.Errorf("error generating keys: %v", err)
-	}
-
-	tokenData := &Payload{
-		Operation: OperationUpload,
-		Name:      "fakepvc",
-		Namespace: "fakenamespace",
-		Resource: metav1.GroupVersionResource{
-			Group:    "",
-			Version:  "v1",
-			Resource: "persistentvolumeclaims",
-		},
-	}
-
-	g := NewGenerator(issuer, key, 5*time.Minute)
-
-	signedToken, err := g.Generate(tokenData)
-
-	if err != nil {
-		t.Errorf("unable to generate token: %v", err)
-	}
-
-	validator := NewValidator(issuer, &key.PublicKey, 0)
-
-	payload, err := validator.Validate(signedToken)
-
-	if err != nil {
-		t.Errorf("unable to verify token: %v", err)
-	}
-
-	if !reflect.DeepEqual(tokenData, payload) {
-		t.Errorf("invalid token payload")
-	}
+	RegisterFailHandler(Fail)
+	RunSpecsWithDefaultAndCustomReporters(t, "Token Suite", reporters.NewReporters())
 }
 
-func TestTokenTimeout(t *testing.T) {
-	issuer := "issuer"
-
-	key, err := generateTestKey()
-	if err != nil {
-		t.Errorf("error generating keys: %v", err)
-	}
-
-	tokenData := &Payload{
-		Operation: OperationUpload,
-		Name:      "fakepvc",
-		Namespace: "fakenamespace",
-		Resource: metav1.GroupVersionResource{
-			Group:    "",
-			Version:  "v1",
-			Resource: "persistentvolumeclaims",
-		},
-	}
+var _ = Describe("Token test", func() {
+	It("Token", func() {
```

Comment:
These 4 are a perfect example of using a DescribeTable in the correct way. Each one basically calls all the same functions but with some different parameters, if you make this a DescribeTable with 4 entries you will reduce the amount of code by like 50%.

Reply:
I've avoided doing this because it felt like I'm about to make an error doing that

Reply:
I am not sure why you feel that way, as you can tell all this code is basically duplicated 4 times with some different parameters. You converted the structs in other places without issues, I don't see why this would be any different.